### PR TITLE
feature: print runtime git commit

### PR DESCRIPTION
**Pull request by hhyasdf**
Pull Request Description
---

### Describe what this PR does / why we need it
Print runtime commit id of Rama in rama-manager and rama-daemon logs. It helps us to find the correct commit id while debugging. 

### Does this pull request fix one issue?

NONE

### Describe how you did it
Link a flag "main.gitVersion" while building binaries, and add logs in rama-daemon and rama-manager.

### Describe how to verify it
Run rama-daemon and rama-manager pod, logs of "starting rama manager with git version: <commit id>" will be printed at the first place. Executing command of "cat /rama/COMMIT_ID" in pod will show commit id too.

### Special notes for reviews

### CODE REVIEW

**Review by mars1024 — CHANGES_REQUESTED** [review]
IMO, I don't think this will help us a lot, because
1. log won't be kept for a long time, it rotates usually
2. commit id is not useful compared with tag, which could be found in image url

**Comment by hhyasdf** [comment]
> IMO, I don't think this will help us a lot, because
> 
> 1. log won't be kept for a long time, it rotates usually
> 2. commit id is not useful compared with tag, which could be found in image url

For the first problem, I add an /rama/VERSION file into the image for long time usage.

For the second problem, image tag sometimes is unreliable. This happens mostly in develop environment, developer sometime don't know the exact code version if they dose not use a official release or have changed the image's tag.

File: Dockerfile.amd64 [hunk 1]
```diff
@@ -11,10 +11,12 @@ RUN export GOCACHE=/tmp && \
     export GOARCH=${ARCH} && \
     export CGO_ENABLED=0 && \
     export GOOS=linux && \
+    export GIT_VERSION=`git rev-parse --short HEAD 2>/dev/null` && \
 	go build -o dist/images/rama -ldflags "-w -s" -v ./cmd/cni && \
-	go build -o dist/images/rama-daemon -ldflags "-w -s" -v ./cmd/daemon && \
-    go build -o dist/images/rama-manager -v ./cmd/manager && \
-    go build -o dist/images/rama-webhook -v ./cmd/webhook
+	go build -ldflags "-w -s -X \"main.gitVersion=`echo $GIT_VERSION`\" " -o dist/images/rama-daemon -v ./cmd/daemon && \
+    go build -ldflags "-X \"main.gitVersion=`git rev-parse --short HEAD 2>/dev/null`\" " -o dist/images/rama-manager -v ./cmd/manager && \
```
Comment (mars1024) [inline Dockerfile.amd64:17]: pay attention to the indent

File: Dockerfile.amd64 [hunk 2]
```diff
@@ -11,10 +11,12 @@ RUN export GOCACHE=/tmp && \
...
+    go build -o dist/images/rama-webhook -v ./cmd/webhook && \
+    echo $GIT_VERSION > ./VERSION
```
Comment (mars1024) [inline Dockerfile.amd64:19]: VERSION seems puzzled to me, could we use another name?

File: cmd/daemon/daemon.go [hunk 3]
```diff
@@ -30,10 +30,14 @@ import (
 	controllerruntime "sigs.k8s.io/controller-runtime"
 )
 
+var gitVersion string
+
 func main() {
 	klog.InitFlags(nil)
 	defer klog.Flush()
 
+	klog.Infof("Starting rama manager with git version: %v", gitVersion)
```
Comment (mars1024) [inline cmd/daemon/daemon.go:39]: nit: manager --> daemon

File: Dockerfile.amd64 (hunk 1 above)
Comment (hhyasdf) [reply Dockerfile.amd64]: fixed

File: Dockerfile.amd64 (hunk 2 above)
Comment (hhyasdf) [reply Dockerfile.amd64]: change to COMMIT_ID

File: cmd/daemon/daemon.go (hunk 3 above)
Comment (hhyasdf) [reply cmd/daemon/daemon.go]: fixed

**Review by mars1024 — APPROVED** [review]
/lgtm
/approve
thanks!

**Pull request merged**